File: backend/router/group/group.go
package group

import (
    "log"

    "github.com/gin-gonic/gin"

    "openvpn-web-admin/middlewares/auth"
    "openvpn-web-admin/models/group"
    "openvpn-web-admin/utils/errutil"
//    "net/http"
)

var router *gin.RouterGroup

func Init(r *gin.RouterGroup) {
    router = r
//    router.GET("/", auth.CheckSignIn, userinfo)
    router.POST("/get", auth.CheckSignIn, getgroup)
    router.GET("/getall", auth.CheckSignIn, getgroups)
//    router.POST("/adduser", adduser)
}

func getgroup(c *gin.Context) {
    postdata := make(map[string]any)
    c.BindJSON(&postdata)
    groupdata, err := group.GetGroup(postdata["groupname"].(string))
    if err != nil {
        log.Panicln(err)
        return
    }
    if groupdata == nil {
        errutil.AbortAndError(c, &errutil.Err{
            Code: 403,
            Msg: "group not exist",
        })
        return
    }
    c.JSON(200, groupdata)
}

func getgroups(c *gin.Context) {
    groupdatas, err := group.GetGroups()
    if err != nil {
        log.Panicln(err)
        return
    }
    c.JSON(200, groupdatas)
}

File: frontend/src/services/group.js
import axios from '@/plugins/axios.js';

function getgroup(groupname) {
  return axios.post('/group/get', {
    groupname,
  }).then((res) => {
    return {
      status: res.status,
      data: res.data,
    };
  }).catch((err) => {
    return {
      status: err.response.status,
      data: err.response.data.msg,
    };
  });
}

function getgroups() {
  return axios.get('/group/getall');
}

export default {
  getgroup,
  getgroups,
}
